pkg/prom/result.go

```diff
@@ -60,6 +60,18 @@ func ValueFieldDoesNotExistErr(query string, promResponse interface{}) error {
 	return fmt.Errorf("Error parsing Prometheus response: 'value' field does not exist in data result vector. Query: '%s'. Response: '%+v'", query, promResponse)
 }
 
+// VersionResultsChan is a channel for Prometheus major version results
+type VersionResultsChan chan int
```

Comment:
My API request to [`/api/v1/status/buildinfo`](https://prometheus.io/docs/prometheus/latest/querying/api/#build-information) returns very different response contents compared to API requests for [`/api/v1/query`](https://prometheus.io/docs/prometheus/latest/querying/api/#instant-queries).

I didn't want to force my response contents into a `QueryResult` type if it didn't make sense. So I started this new type which is just a Channel for an int representing the Prometheus major version number. I'm starting with a very narrow scope, but open to broadening it if you think it makes sense.

Reply:
No longer relevant. I reverted this commit.